llvm/lib/Target/PowerPC/PPCISelDAGToDAG.cpp

```diff
@@ -7566,6 +7566,100 @@ static void reduceVSXSwap(SDNode *N, SelectionDAG *DAG) {
   DAG->ReplaceAllUsesOfValueWith(SDValue(N, 0), N->getOperand(0));
 }
 
+// Is an ADDI eligible for folding for non-TOC-based local-exec accesses?
+static bool isEligibleToFoldADDIForLocalExecAccesses(SDNode *N,
+                                                     SelectionDAG *DAG,
+                                                     SDValue ADDIToFold) {
+  const PPCSubtarget &Subtarget =
+      DAG->getMachineFunction().getSubtarget<PPCSubtarget>();
+  // This optimization is only performed for non-TOC-based local-exec accesses.
+  if (!Subtarget.hasAIXSmallLocalExecTLS())
+    return false;
+
```

Comment:
I do not think putting the code in the function `isEligibleToFoldADDIForLocalExecAccesses`  is efficiently 

 ```
    const PPCSubtarget &Subtarget =
      DAG->getMachineFunction().getSubtarget<PPCSubtarget>();
  // This optimization is only performed for non-TOC-based local-exec accesses.
  if (!Subtarget.hasAIXSmallLocalExecTLS())
      return false;
```

I will put the code in front of function as void `PPCDAGToDAGISel::PeepholePPC64() `

as 
void PPCDAGToDAGISel::PeepholePPC64() {
```
const PPCSubtarget &Subtarget =
      DAG->getMachineFunction().getSubtarget<PPCSubtarget>();
  // This optimization is only performed for non-TOC-based local-exec accesses.
bool HasAIXSmallLocalExecTLS = Subtarget.hasAIXSmallLocalExecTLS() ;
.....

if (HasAIXSmallLocalExecTLS) 
   foldADDIForLocalExecAccesses();
   
       } else if (Offset != 0) {
        if (HasAIXSmallLocalExecTLS && isEligibleToFoldADDIForLocalExecAccesses(N, CurDAG, Base)) {
        
```


to avoid multi time get and check Subtarget.hasAIXSmallLocalExecTLS() in the loop of function `PeepholePPC64`